Theory assessment/Theory_assessment_Software_Engineering.md

```diff
@@ -0,0 +1,150 @@
+```
+class Animals:
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        
+    def lives_in_farm(self):
+        return True
+            
+class Dog(Animals):
+    def __init__(self, name, age):
+        super().__init__(name, age)
+        
+    def lives_in_farm(self):
+        return False
+
+dog_Mika=Dog('Mika', 3)
+print(dog_mika.name, dog_mika.age, dog_mika.lives_in_farm())
+```
+### 3. What's inheritance in OOP?
+A: Inheritance in OOP means that we can transfer methods/properties from one class to another. It helps with code reading, reusing and scaling and decreases repetition. Also better and easier to find errors.
+```
+class Animals:
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        
+    def lives_in_farm(self):
+        return True
+            
+class Cow(Animals):
+    def __init__(self, name, age):
+        super().__init__(name, age)       
+          
+cow_alma=Cow('Alma', 12)
+print(cow_alma.name, cow_alma.age, cow_alma.lives_in_farm())
+```
+### 4. If you had to make a program that could vote for the top three funniest people in the office, how would you do that? How would you make it possible to vote on those people?
+- APPROACH: I would firstly make plan, determine needed tools and program architecture. I think in this case it would be great if there would be e-mail link for voting, so everybody could vote one time and they could be identified by e-mail, so we could keep track how has voted and send reminders. The program has multiple choice answer questions, where you can only vote for one person. If there are a lot of different departments, then I would make extra choice, to see who are in which department. After voting is ended, you can see who are three people who got most votes. When somebody is in a tie, then both names are shown.
+- KEY REQUIREMENTS:
+  - Every employee can vote one time and for one person.
+  - They can see later the result if voting is ended.
+  - Getting notification if results are out. 
+- MAIN CONSIDERATIONS:
+  - How to make sure that person votes only one time. 
+  - Can person change their vote until the end, just in case they choose wrong person.
+  - If there are a lot of people/similar names in list, then how to make system easy for voting, so there would not be confusion. 
+- BIGGEST/COMMON PROBLEMS:
+    - Confusion with names or people, if there is a lot of employees.
+    - People wanting to change their votes.
+- COMPONENTS/TOOLS:
+    - SQL for holding data and e-mails - who have voted.
+    - PYTHON for back-end logic.
+    - HTML and CSS for front-end.
+    - API for e-mail notifications.
+
+### 5. What's the software development cycle?
+A: Software development cycle known as SDLC, is needed to deliver expected goals to client. It has 6 steps:
```

Comment:
6/10 (-2)The deploying part should be on a separate stage. it is not part of Maintenance. (-2) The quality of the description can be better. I'd like to see how or which tools are normally being used for each cycle, who is involved there and what is the result of each stage.